### PR TITLE
feat: State with account status

### PR DESCRIPTION
**Pull request by rakita**
Model account states to cover all possible state that transaction can change it

Model can be found here
https://mermaid.live/edit#pako:eNqNU7FOwzAQ_RXLI2qQysDgAQnRDEhQITFiBss-Uku1XTkXIKr677gxIa5jlWZInHvv3t3L5fZUOgWU0RYFwkqLxgtTfd5wS8K1gha960GRqrqb3tbwRRix4S49hLRIftgI2-TUOfR7jsBRqcAfw-GZ089UydA8-aT93M-FzPtGaJuRhthMcK5U7Dwv_eSEAlWbHfb148vydjn_Zv9qJk2W5aaxMjLMncgtCH99LufPUoTXDutv3aK2TQmelVGno0pYowVGZOry7ep9gCOzFEtaKMGJAbqgBrwRWoUffX8kc4obMMApC0cFH6LbIqfcHgJVdOheeyspQ9_BgnY7Na3GGASl0fnnuDvDCh1-AFmUEJU

And create cached/transition/bundle states that can easily create reverts on those states.

States interconnection can be seen here:
```
EVM State
(It has both original/present storage and new account and storage acts as a access list.)
|
|
V
[Cache State] Fetches data from database and get updated from EVM state. Allows us fewer layer to fetch data from database.
|        \
|         \
|          [Transitions State] contains changes related to transition (Mostly block). It has original account/storage/status (Needed for Loaded account). It is used for creating Reverts. 
|          |
|          V
|          [Bundle state] (It has Reverts and plain state, Original storage is not needed) One of reasons why this is the case, is because when reverting of canonical chain we can't get previous storage value. And it is not needed.
|       /
|    /
v
database
```

this is still WIP!

### CODE REVIEW

**Comment by rakita** [comment]
Okay first update.

I bootstrapped the account state (with ducktape) to work with reth pipeline execution stage. And synced the first 3M sepolia blocks (i didn't pass Merkle stages so there is still some debugging, and revert and receipts are not yet saved).
 
Code is now easier to navigate and the creation of changesets (I call them Reverts) is now simpler, so the main point of doing this seems to be on right path.

The previous run i had 6min 42s and now i am getting 5m 30s, this still needs to be verified and receipts/changesets (although generated) are still not written to db.

So last thing that I wanted to see is how much time is spent on post-executing and what time we would save if it was done in parallel.
Times go something like this:
```
2023-06-13T15:14:52.263843Z DEBUG Executing range start=2900030 end=3500000
.
2023-06-13T15:15:22.690379Z  INFO Status connected_peers=6 stage=Execution checkpoint=2900029
T - 56.940997292s Executed
  I - 2.132384515s Block Read time
  F - 205.189225ms evm init
  F - 51.288308458s evm exec
  F - 196.752879ms evm finish
 X - 51.801919484s execution time
 X - 1.960304335s receipt root time
   E - 276.777256ms apply evm state
   E - 166.733308ms apply transition state
   E - 468.53844ms merge time
   DB - 228.100688ms get account
   DB - 686.969336ms get storage
   DB - 6.321257ms get code
  I - 260.853584ms Take State
2023-06-13T15:15:50.171638Z  INFO Wrote state took=706.247083ms
.
2023-06-13T15:15:50.407192Z DEBUG Executing range start=3000031 end=3500000
```

I used simple `Instant::now()` and added times. Full execution of 100k blocks from 2_900_00 to 3_000_000 took ~55s and of those ~51s was in revm call/create and interesting things is that revm db access (both cache and disk) took just  ~900ms that would mean that access to the database not the bottleneck of execution but revm is. 
In summary:
* ~900ms db access from revm.
* ~900ms Creation of changes and reverts.
* ~1.9s receipts root
* ~51s revm execution.
* ~700ms writing to db.

This means after this get finished, most of time in execution would be just spent in evm execution loop.

Exec timestamps can be found here:
https://github.com/bluealloy/revm/blob/6dbed63c375344fb66114690a0c9150f5f98ae6d/crates/revm/src/evm_impl.rs#L115-L156

Commit from reth size is here: https://github.com/paradigmxyz/reth/commit/657aed4b63939c9f7f1aba4ef43e8b34a66e7239

**Comment by rakita** [comment]
Adding account status that the states are modeled from: https://mermaid.live/edit#pako:eNqdU7FOwzAQ_RXLI2qpysCQAQnRDpUoQmLEDId9TSMl58q5IKKq_45xGnAat0Jkct69e37v5NtLbQ3KTNYMjIsCcgfV9ONGkfDf69WbmE7vxKMFg2ZZ7bhdrp7nt_NU-cny8rOouaA8Ve6wETUwVrTGyrr2YQuUo8jEO5RAGoUOwEwp0pYcaBbaoTc6UIt8BbUF1uxsi8YLhVhClwju-lLP0EHMTBvUlgd2FA0tC7LkSRRYI7lOxiSD1mwd5D0yao3D9frC9GBH7-X_w-_PYXaxk9koY9d8MprEtOKbu3mNtc4lT6gvhuZ_fpOpj-BfqPc5FHRCTIrFeRweX0AcyZeo3qA7EQsXnLEnJ7JCV0Fh_C7uvxuV5C1WqGTmjwY30JSspKKDp0LD9qUlLTN2DU5kszO_29uDaAo_0HW33mHLD19F7VNm

**Comment by rakita** [comment]
tests were checked locally.

**Review by rakita — COMMENTED** [review]
This PR synced on 17M blocks so we can assume initial sync works as expected.

There are few more bugs found related to reverts (changesets in reth) that still need to be addressed

We are merging this code and will iterate on it so we can unblock other teams that depend on revm.
